spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobService.java

```diff
@@ -225,9 +225,7 @@ public int stopAll() {
 		Collection<JobExecution> result = jobExecutionDao.getRunningJobExecutions();
 		for (JobExecution jobExecution : result) {
 			try {
-				jobExecution.getStepExecutions().forEach(StepExecution::setTerminateOnly);
-				jobExecution.setStatus( BatchStatus.STOPPING);
-				jobRepository.update(jobExecution);
+				stop(jobExecution.getId());
```

Comment:
Because we already have loaded the execution from the DB it would be an improvement to create a private `stop(JobExecution)` and then have both this and the one in the singular `stop(Long)` call into it.